src/Rings/mpolyquo-localizations.jl

```diff
@@ -1696,8 +1696,9 @@ function saturation(I::IdealType, J::IdealType) where {IdealType<:Union{MPolyQuo
   A = base_ring(I)
   A === base_ring(J) || error("ideals must lie in the same ring")
   R = base_ring(A)
-  Ip = saturated_ideal(I)
-  Jp = saturated_ideal(I)
+
+  I_sat = ideal(base_ring(R), normal_form(gens(saturated_ideal(I)), modulus(R)))
+  J_sat = ideal(base_ring(R), normal_form(gens(saturated_ideal(J)), modulus(R)))
```

Comment:
Could you briefly explain what the problem was here? From your mail I understood that there might be something wrong with the implementation of `saturated_ideal`.